who_depends_this_lib

```diff
@@ -64,14 +65,24 @@ def extract_package(pkg):
   finally:
     shutil.rmtree(d, onerror=partial(handle_rmtree_error, d))
 
+@contextlib.contextmanager
+def tarfile_open_zstd_compat(name):
+  if name.endswith('.zst'):
+    dctx = zstandard.ZstdDecompressor()
+    with open(name, 'rb') as f:
+      with dctx.stream_reader(f) as reader:
+        yield tarfile.open(mode='r|', fileobj=reader)
+  else:
+    yield tarfile.open(name)
```

Comment:
It's wrong to yield contextmanagers; they need to be used with `with`, or its `__exit__` method won't be called.

Also, you can `with` several things in one line like `with open(name, 'rb') as f, dctx.stream_reader(f) as reader, tarfile.open(mode='r|', fileobj=reader) as tar`.

Reply:
Thank you. I really need to study again how `with` works :sweat_smile: